File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0024/24.swift
class Solution {
    func swapPairs(_ head: ListNode?) -> ListNode? {
        let dummy = ListNode()
        dummy.next = head
        var pre = dummy
        while ((pre.next != nil) && ((pre.next?.next) != nil)) {
            let a = pre.next;
            let b = pre.next?.next;
            let tmp = b?.next;
            pre.next = b;
            b?.next = a;
            a?.next = tmp;
            pre = a ?? pre;
        }
        return dummy.next;
    }
}

File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0024/24.js
var swapPairs = function (head) {
  const dummy = new ListNode();
  dummy.next = head;
  let pre = dummy;
  while (pre.next && pre.next.next) {
    const a = pre.next;
    const b = pre.next.next;
    const tmp = b.next;
    pre.next = b;
    b.next = a;
    a.next = tmp;
    pre = a;
  }
  return dummy.next;
};
